docs/conf.py

```diff
@@ -13,6 +13,35 @@
 import os
 import sys
 import subprocess
+import pkgutil
+import re
+
+
+def generate_hoc_domain_from_py():
+    py_domain = pkgutil.get_data('sphinx.domains', 'python.py').decode('utf-8')
+
+    hoc_dict = {
+        " (%s ": " (HOC %s ",
+        "'%s (": "'%s (HOC ",
+        "'built-in function'": "'HOC built-in function'",
+        "(built-in variable)": "(HOC built-in variable)",
+        "'python": "'hoc",
+        "python_": "hoc_",
+        "Python": "HOC",
+        "Py": "HOC",
+        "'py": "'hoc",
+        "py_": "hoc_",
+    }
+    # Create a regular expression  from the dictionary keys
+    regex = re.compile("(%s)" % "|".join(map(re.escape, hoc_dict.keys())))
+
+    # For each match, look-up corresponding value in dictionary
+    hoc_domain = regex.sub(lambda mo: hoc_dict[mo.string[mo.start():mo.end()]], py_domain)
+
+    with open('hocdomain.py', 'w') as hoc_py:
+        hoc_py.writelines(hoc_domain)
+
+generate_hoc_domain_from_py()
```

Comment:
Still not decided on the strategy here. We can

- generate the HOC domain with every Sphinx build (like we do here), or
- generate it separately and commit it to the repo

I'm inclining more towards the latter since it might require less attention maintenance wise. In this case we can move `generate_hoc_domain_from_py():` somewhere else and document it for future use.